File: client/src/api/items/create.ts
import { Headers } from "@/types/global";
import { api, api_v1_safe } from "../api";

export class APICreateItem {
    private token: string
  
    constructor(token: string) {
        this.token = token;
    }
  
      public async execute(safe_id: string, body: object): Promise<any> {
          return new Promise((resolve, reject) => {
            let headers: Headers = {
                'Authorization': `Bearer ${this.token}`,
                'Content-Type': 'application/json'
            }
            api.post(`${api_v1_safe}/${safe_id}/item`, body, {
                headers: headers,
                })
                .then((response: any) => {
                resolve(response);
                })
                .catch((error) => {
                reject(error.response);
            });
        });
    }
}

File: tests/api/item/create.py
from typing import Literal
from fastapi.testclient import TestClient
from mindfulguard.__main__ import app

class ItemCreateApi:
    def __init__(
        self,
        path: str,
    ) -> None:
        self.__client = TestClient(app)
        self.__path: str=  path

    def execute(self, headers: dict[str, str], body: dict[str, str], safe_id: str):
        return self.__client.post(f"{self.__path}/{safe_id}/item", json = body, headers = headers)